### PR TITLE
Adding an example setup and VERSION file

### PR DESCRIPTION
**Pull request by jmlarson1**

### CODE REVIEW

**Comment by jmlarson1** [comment]
@jared321 What do you think of this setup.py file?

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import codecs
+import os.path
+from setuptools import setup
+
+here = os.path.abspath(os.path.dirname(__file__))
```
Comment (jared321) [inline setup.py:5]: If you are interested in modernizing, try
```
from pathlib import Path

_FILE_PATH = Path(__file__).parent
```

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+def readme():
+    fname = os.path.join(here, "README.md")
+    with codecs.open(fname, encoding="utf8") as fptr:
```
Comment (jared321) [inline setup.py:10]:
```
fname = _FILE_PATH.joinpath("README.md")
```

with similar below.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+def version():
+    fname = os.path.join(here, "VERSION")
+    with open(fname, "r") as fptr:
+        return fptr.read().strip()
+
+
+reqs_list = ["numpy", "scipy"]
+
+pkg_dict = {"ibcdfo": ["pounders/py/*", "pounders/py/tests/regression_tests/*"]}
```
Comment (jared321) [inline setup.py:22]: I used these 
* `PkgData/*` - add in all internal data files whose contents could be loaded through the package interface
* `TestData/*` - add in all internal data files whose contents are consumed by the testing built into the package.

Do you need to declare your folders as part of this package?

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    packages=["ibcdfo"],
+    package_data=pkg_dict,
+    url="https://github.com/POptUS/IBCDFO/",
+    license="MIT License",
+    description="Interpolation-Based Composite Derivative-Free Optimization",
+    long_description=readme(),
+    setup_requires=[],
+    install_requires=reqs_list,
+    tests_require=test_list,
+    # test_suite='ibcdfo.test_suite',
```
Comment (jared321) [inline setup.py:42]: I believe that `test_suite` was part of running testing via `python setup.py test`, which will be deprecated.  I get messages suggesting I use `vox` instead.  That's on my TODO list.

 It makes sense that you are commenting it out then.  Not sure if you need `tests_require` then.

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        "Development Status :: 4 - Beta",
+        "License :: OSI Approved :: MIT License",
+        "Programming Language :: Python",
+        "Intended Audience :: Developers",
+        "Intended Audience :: Science/Research",
+        "Natural Language :: English",
+        "Operating System :: POSIX :: Linux",
+        "Operating System :: Unix",
+        "Operating System :: MacOS",
+        "Operating System :: Microsoft :: Windows :: Windows 7",
```
Comment (jared321) [inline setup.py:54]: I don't keep track of Windows, but I think that they are well past 7.

**Comment by jmlarson1** [comment]
@jared321 Thanks. Most of your suggestions have been implemented. I don't understand the recommendations for changes for the TestData and PkgData fields. Can you clarify?

**Comment by jared321** [comment]
I included into all my python packages those two folders that I mentioned.  I didn't populate those with code, but rather with data files needed internally by the package for various reasons.  For instance, maybe I include some `*.mat` or `*.h5` files.

Are you doing something similar or something else that is actually necessary in your use of `pkg_dict`?  My packages were always flat (e.g., I never had to do something like `import ibcdfo.pounders`, so I'm not sure if there are additional uses for it.

**Review by mmenickelly — APPROVED** [review]

File: pounders/py/general_h_funs.py [hunk 6]
```diff
@@ -45,10 +45,12 @@ def emittance_h(F):
 
 
 def squared_diff_from_mean(Cres, Gres, Hres):
-    # Combines models for the following h function
-    #    h = @(F)sum((F - 1/m*sum(F)).^2) - alpha*(1/m*sum(F))^2
-    # That is, the objective is to have the vector close to it's mean, and have
-    # a small mean (penalized by alpha)
+    """
+    Combines models for the following h function
+       h = @(F)sum((F - 1/m*sum(F)).^2) - alpha*(1/m*sum(F))^2
```
Comment (jared321) [inline pounders/py/general_h_funs.py:50]: Is this MATLAB notation in a python docstring?

**Comment by jared321** [comment]
I just did a final review.  It looks generally good to me with the understanding that we continue to evolve and improve the python package and the use of this as testing is integrated into the package.

**Review by jared321 — APPROVED** [review]

File: pounders/py/general_h_funs.py (hunk 6 above)
Comment (jmlarson1) [reply pounders/py/general_h_funs.py:50]: It is, but it should be legible to most users interested in such functionality.

**Comment by jmlarson1** [comment]
@jlnav Note that @jared321 changed the `ci.yml` file. Please let us know if anything seems out of place.

**Pull request merged**